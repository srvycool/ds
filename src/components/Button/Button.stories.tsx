import React from 'react';
import { action } from '@storybook/addon-actions';
import { Button } from './Button';

export default {
  title: 'Button',
  component: Button,
};

export const Primary: React.FC = () => (
  <Button onClick={action('clicked')}>Button</Button>
);
