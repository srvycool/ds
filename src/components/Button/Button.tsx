import styled from 'styled-components';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ButtonProps {}

export const Button = styled.button<ButtonProps>`
  height: 44px;
  background-color: ${(props) => props.theme.colors.primary500};
  color: ${(props) => props.theme.colors.grey000};
  font-family: ${(props) => props.theme.fonts.secondary};
  border: none;
  font-size: ${(props) => props.theme.fontSizes[1]};
  padding: 0 ${(props) => props.theme.space.m};
  border-radius: 22px;
  cursor: pointer;
`;
