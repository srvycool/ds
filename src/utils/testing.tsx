import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
