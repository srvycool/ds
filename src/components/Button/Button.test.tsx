import React from 'react';
import { render } from '../../utils/testing';
import { Button } from './Button';

describe('Button', () => {
  it('should match snapshot', () => {
    const { container } = render(<Button>Click me</Button>);
    expect(container).toMatchSnapshot();
  });
});
