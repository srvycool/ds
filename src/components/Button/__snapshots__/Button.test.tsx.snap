// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
.c0 {
  height: 44px;
  background-color: #DA4167;
  color: #FFFFFF;
  font-family: Montserrat;
  border: none;
  font-size: 16px;
  padding: 0 16px;
  border-radius: 22px;
  cursor: pointer;
}

<div>
  <button
    class="c0"
  >
    Click me
  </button>
</div>
`;
